onnxscript/ir/passes/common/unused_removal.py

```diff
@@ -186,3 +186,33 @@ def call(self, model: ir.Model) -> ir.passes.PassResult:
                 modified |= self._process_graph_like(function, used_domains=set())
 
         return ir.passes.PassResult(model, modified=modified)
+
+
+class RemoveUnusedMetadataAndDocStringPass(ir.passes.InPlacePass):
+    def call(self, model: ir.Model) -> ir.passes.PassResult:
```

Comment:
I think there is also model metadata and docstring

Reply:
As well as other objects, if IIRC (tensors?). I wonder, though, if removing model metadata is a bit extreme. Of course, it is up to the user who calls this ... but perhaps there should be options for users to control this ... which types of objects to remove metadata from.

Reply:
For reference, ORT manages to remove graphs and nodes only for model shipping.

Reply:
https://github.com/microsoft/onnxruntime/blob/d7c688e15c1dc40f57140bff08c78e01a88b19fc/onnxruntime/python/tools/transformers/dynamo_onnx_helper.py#L176-L179

Reply:
Adding onto this, #2167 for example, could provide a case where metadata_props could be used to propagate model properties (max_seq_len in this case). If we remove all model metadata by default, is there an alternative method to propagate model specific parameters or properties?

Reply:
This pass happens in the end of optimization call. Could you confirm if this works in the case?

Reply:
Sorry, it's not used by default. It's an option for users.